src/main/kotlin/minesweeper/entity/Coordinate.kt

```diff
@@ -5,4 +5,49 @@ data class Coordinate(val x: Int, val y: Int) {
         require(x >= 0) { "x는 0보다 커야합니다." }
         require(y >= 0) { "y는 0보다 커야합니다." }
     }
+
+    fun adjacentCoordinates(): List<Coordinate> {
+        val directions = listOf(-1, 0, 1)
+        return directions.flatMap { deltaX ->
+            getAdjacentCoordinatesForRow(deltaX, directions)
+        }
+    }
+
+    private fun getAdjacentCoordinatesForRow(
+        deltaX: Int,
+        directions: List<Int>,
+    ): List<Coordinate> {
+        return directions.mapNotNull { deltaY ->
+            when {
+                deltaX == 0 && deltaY == 0 -> null
+                else -> {
+                    val adjacentX = x + deltaX
+                    val adjacentY = y + deltaY
+                    if (adjacentX >= 0 && adjacentY >= 0) {
+                        Coordinate(adjacentX, adjacentY)
+                    } else {
+                        null
+                    }
+                }
+            }
+        }
```

Comment:
지금은 map block의 조건식이 조금 복잡하게 느껴져서 한눈에 파악하기가 조금 어렵게 느껴지는 것 같아요. 🤔

deltaX, deltaY가 0인 조건은 애초에 필터링해버리면 조금 더 mapping 로직이 깔끔해질 것 같아요.

```suggestion
        return directions
            .filterNot { deltaY -> deltaX == 0 && deltaY == 0 }
            .map { deltaY ->
                Coordinate(x + deltaX, y + deltaY)
            }
```